src/Scripting/VisualBasic/Hosting/ObjectFormatter/VisualBasicPrimitiveFormatter.vb

```diff
@@ -0,0 +1,81 @@
+' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.Scripting.Hosting
+Imports Microsoft.CodeAnalysis.Scripting.Hosting.ObjectFormatterHelpers
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Scripting.Hosting
+
+    Public Class VisualBasicPrimitiveFormatter
+        Inherits CommonPrimitiveFormatter
+
+        Protected Overrides ReadOnly Property NullLiteral As String
+            Get
+                Return "Nothing"
+            End Get
+        End Property
+
+        Protected Overrides Function FormatLiteral(value As Boolean) As String
+            Return ObjectDisplay.FormatLiteral(value)
+        End Function
+
+        Protected Overrides Function FormatLiteral(value As Date) As String
+            Return ObjectDisplay.FormatLiteral(value)
+        End Function
+
+        Protected Overrides Function FormatLiteral(value As String, useQuotes As Boolean, escapeNonPrintable As Boolean, Optional numberRadix As Integer = NumberRadixDecimal) As String
```

Comment:
Per offline discussion, since we're permissive about other options (like useHexadecimal), perhaps it's okay to be permissive here (and it would be a bit odd to only validate *some* of the arguments), but we should make sure that ObjectDisplay does something reasonable, even for cases that "don't make sense" (and make sure there are tests).